dataline-workers/src/test/java/io/dataline/workers/singer/SingerTapFactoryTest.java

```diff
@@ -0,0 +1,164 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.workers.singer;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Lists;
+import io.dataline.config.SingerCatalog;
+import io.dataline.config.SingerColumn;
+import io.dataline.config.SingerColumnMap;
+import io.dataline.config.SingerMessage;
+import io.dataline.config.SingerStream;
+import io.dataline.config.SingerTableSchema;
+import io.dataline.config.SingerType;
+import io.dataline.config.StandardDiscoverSchemaInput;
+import io.dataline.config.StandardTapConfig;
+import io.dataline.workers.DefaultSyncWorker;
+import io.dataline.workers.InvalidCredentialsException;
+import io.dataline.workers.JobStatus;
+import io.dataline.workers.OutputAndStatus;
+import io.dataline.workers.StreamFactory;
+import io.dataline.workers.TestConfigHelpers;
+import io.dataline.workers.WorkerUtils;
+import io.dataline.workers.process.ProcessBuilderFactory;
+import io.dataline.workers.protocol.singer.MessageUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SingerTapFactoryTest {
+  private static final String IMAGE_NAME = "hudi:latest";
+  private static final String JOB_ROOT_PREFIX = "workspace";
+  private static final String DISCOVER_SCHEMA_JOB_ROOT_PREFIX = "discover";
+  private static final String CONFIG_JSON_FILENAME = "tap_config.json";
+  private static final String CATALOG_JSON_FILENAME = "catalog.json";
+  private static final String STATE_JSON_FILENAME = "input_state.json";
+  private static final String TABLE_NAME = "user_preferences";
+  private static final String COLUMN_NAME = "favorite_color";
+
+  private Path jobRoot;
+  private Path discoverSchemaJobRoot;
+  private Path errorLogPath;
+
+  @BeforeEach
+  public void setup() {
+    try {
+      jobRoot = Files.createTempDirectory(JOB_ROOT_PREFIX);
+      discoverSchemaJobRoot = jobRoot.resolve(DISCOVER_SCHEMA_JOB_ROOT_PREFIX);
+      errorLogPath = jobRoot.resolve(DefaultSyncWorker.TAP_ERR_LOG);
+
+    } catch (IOException e) {
```

Comment:
Don't catch in tests, it makes the test unnecessarily verbose, just add throws to the prototype